tests/test_models.py

```diff
@@ -1,38 +1,186 @@
+        fake_wishlist = WishlistFactory()
+        # pylint: disable=unexpected-keyword-arg
+        wishlist = Wishlist(
+            customer_id=fake_wishlist.customer_id,
+            wishlist_name=fake_wishlist.wishlist_name,
+            created_date=fake_wishlist.created_date,
+        )
+        self.assertIsNotNone(wishlist)
+        self.assertEqual(wishlist.wishlist_id, None)
+        self.assertEqual(wishlist.customer_id, fake_wishlist.customer_id)
+        self.assertEqual(wishlist.wishlist_name, fake_wishlist.wishlist_name)
+        self.assertEqual(wishlist.created_date, fake_wishlist.created_date)
+
+    def test_add_a_wishlist(self):
+        """It should Create a Wishlist and add it to the database"""
+        wishlists = Wishlist.all()
+        self.assertEqual(wishlists, [])
+        wishlist = WishlistFactory()
+        wishlist.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertIsNotNone(wishlist.wishlist_id)
+        wishlists = Wishlist.all()
+        self.assertEqual(len(wishlists), 1)
+
+    def test_read_wishlist(self):
+        """It should Read a Wishlist"""
+        wishlist = WishlistFactory()
+        wishlist.create()
+
+        # Read it back
+        found_wishlist = Wishlist.find(wishlist.wishlist_id)
+        self.assertEqual(found_wishlist.wishlist_id, wishlist.wishlist_id)
+        self.assertEqual(found_wishlist.customer_id, wishlist.customer_id)
+        self.assertEqual(found_wishlist.wishlist_name, wishlist.wishlist_name)
+        self.assertEqual(found_wishlist.created_date, wishlist.created_date)
+
+    def test_update_wishlist(self):
+        """It should Update a Wishlist"""
+        wishlist = WishlistFactory(wishlist_name="change name")
+        wishlist.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertIsNotNone(wishlist.wishlist_id)
+        self.assertEqual(wishlist.wishlist_name, "change name")
+
+        # Fetch it back
+        wishlist = Wishlist.find(wishlist.wishlist_id)
+        wishlist.wishlist_name = "name is changed"
+        wishlist.update()
+
+        # Fetch it back again
+        wishlist = Wishlist.find(wishlist.wishlist_id)
+        self.assertEqual(wishlist.wishlist_name, "name is changed")
+
+    def test_delete_an_wishlist(self):
+        """It should Delete a Wishlist from the database"""
+        wishlists = Wishlist.all()
+        self.assertEqual(wishlists, [])
+        wishlist = WishlistFactory()
+        wishlist.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertIsNotNone(wishlist.wishlist_id)
+        last_id = wishlist.wishlist_id
+        wishlists = Wishlist.all()
+        self.assertEqual(len(wishlists), 1)
+        wishlist = wishlists[0]
+        self.assertEqual(wishlist.wishlist_id, last_id)
+        wishlist.delete()
+        wishlists = Wishlist.all()
+        self.assertEqual(len(wishlists), 0)
```

Comment:
I think we could make this simpler by .create(), then grab the auto-generated ID.  call .delete() then assert that a .find() with the previous ID does not return a record.

Reply:
I'd argue this is simpler because it does not require looking up a specific id.
The table is empty at the start of each test from the setUp method, so the length should always be 0.

I'll add a .find() assert as well to be more inclusive in this test case, (deleting an entry from a list of 2).